lib/http1.c

```diff
@@ -158,6 +153,14 @@ static void cleanup_connection(struct st_h2o_http1_conn_t *conn)
         /* TODO use lingering close */
         close_connection(conn, 1);
         return;
+    } else {
```

Comment:
I think we do not need to have an else clause here, because the `if` above uses early return.